tests/test_align.py

```diff
@@ -0,0 +1,180 @@
+   -1.2711   -0.8618    0.6186 H   0  0  0  0  0  0  0  0  0  0  0  0
+   -0.1781   -0.6848    1.4006 H   0  0  0  0  0  0  0  0  0  0  0  0
+  1  2  1  0
+  1  3  1  0
+  1 13  1  0
+  1 14  1  0
+  2  6  1  0
+  2 15  1  0
+  2 16  1  0
+  3  4  1  0
+  3  7  1  0
+  3  8  1  0
+  4  5  1  0
+  4  9  1  0
+  4 10  1  0
+  5  6  1  0
+  5 12  1  0
+  5 17  1  0
+  6 11  1  0
+  6 18  1  0
+M  END
+$$$$""",
+        removeHs=False,
+    )
+    return mol_a, mol_b
+
+
+def get_core(mol_a, mol_b, threshold=2.0):
+    """Simple utility that finds a core by using the conformers with a threshold"""
+    mcs_result = mcs(mol_a, mol_b, threshold=threshold)
+    query = Chem.MolFromSmarts(mcs_result.smartsString)
+    return get_core_by_mcs(mol_a, mol_b, query, threshold=threshold)
+
+
+@pytest.mark.nogpu
+def test_align_mols_by_core():
+    """Uses a core to align two compounds. Uses two cyclohexanes with different conformers as the test case in the case of
+    finding the best MCS"""
+
+    mol_a, mol_b = get_cyclohexanes_different_confs()
+
+    assert mol_a.GetNumAtoms() == mol_b.GetNumAtoms()
+
+    core = get_core(mol_a, mol_b)
+
+    # The difference in conformer prevents a complete mapping
+    assert len(core) != mol_a.GetNumAtoms()
+
+    hydrogenless_mol_a = Chem.RemoveHs(deepcopy(mol_a))
+    hydrogenless_mol_b = Chem.RemoveHs(deepcopy(mol_b))
+
+    restraint_core = get_core(hydrogenless_mol_a, hydrogenless_mol_b)
+
+    assert len(restraint_core) == 6  # Number of carbons in mol
+
+    ff = Forcefield.load_from_file(DEFAULT_FF)
+
+    conf_a, conf_b = align.align_mols_by_core(mol_a, mol_b, restraint_core, ff)
+
+    set_romol_conf(mol_a, conf_a)
+    set_romol_conf(mol_b, conf_b)
+
+    core = get_core(mol_a, mol_b)
+
+    # The mapping should cover the entirety of the two compounds
+    assert len(core) == mol_a.GetNumAtoms()
+
+
+def test_align_core_different_size_mols():
+    """Verifies that this works as expected if the size of the molecules aren't the same."""
+    mol_a = Chem.AddHs(Chem.MolFromSmiles("c1ccccc1"))  # benzene
+    mol_b = Chem.AddHs(Chem.MolFromSmiles("c1ccccc1-c2ccccc2"))  # bi-phyenl
+    seed = 2022
+    AllChem.EmbedMolecule(mol_a, randomSeed=seed)
+    AllChem.EmbedMolecule(mol_b, randomSeed=seed)
+
+    # Compounds don't overlap enough to build a core
+    with pytest.raises(AtomMappingError):
+        get_core(mol_a, mol_b)
+
+    # Align based on the first ring in each
+    core = np.zeros((6, 2), dtype=int)
+    core[:, 0] = np.arange(6, dtype=int)
+    core[:, 1] = np.arange(6, dtype=int)
+
+    ff = Forcefield.load_from_file(DEFAULT_FF)
+
+    conf_a, conf_b = align.align_mols_by_core(mol_a, mol_b, core, ff)
+
+    set_romol_conf(mol_a, conf_a)
+    set_romol_conf(mol_b, conf_b)
+
+    core = get_core(mol_a, mol_b)
+
+    # The mapping should cover the entirety of the benzene minus the hydrogen covering the carbon linker
+    assert len(core) == mol_a.GetNumAtoms() - 1
```

Comment:
Testing coverage: Can we add a test of `atom_mapping.get_core_with_alignment` to `test_align.py` or `test_atom_mapping.py`?

(As of https://github.com/proteneer/timemachine/commit/68690b35b7e773a3d11f3738c5015638be2187f9 , getting a`ValueError: cannot extract desired type from sequence`  at [L183](https://github.com/proteneer/timemachine/pull/813/files#diff-79ea6707b80f02c8fa97a1b42d9e52116bd31149cce41220eefbb0ca7a56191cR183) 
, even though `align_mols_by_core` and `set_romol_conf` look good individually)

Reply:
Where do you see that error? Testing with the cyclohexane case doesn't seem to raise the error locally.

Reply:
Fixed here: [6cce4b0](https://github.com/proteneer/timemachine/pull/813/commits/6cce4b0e5132d23081fe2e3e62a684c1ed141804)

The test fails without the cast to float64 in `set_romol_conf`